### ISSUE TITLE
Determine new sources of data

### ISSUE THREAD

**Issue by rsangole** [rsangole/capstone_project#6]
@andrew3cooper @tedinciong @stephenhage @kapelinskim6 @racheldudle 

Please list all the sources of new data here. Include a quick description, and a link to the data, and rank how likely it is to be useful.

We can use this as a checklist of what we've investigated and mark them complete one at a time.

**Comment by rsangole** [comment]
[ ] Census American Community Survey - #3 
[ ] Weather data outside NOAA - #4 
[ ] Lakes/still water locations
[ ] Senior home locations
[ ] Hospital locations
[ ] School locations

**Comment by rsangole** [comment]
@andrew3cooper et al:

I find data like [this one](https://data.cityofchicago.org/Buildings/Vacant-and-Abandoned-Buildings-Violations/kc9i-wq85/data) . These may be good substitutes for hypotheses we present ourselves... how do abandoned locations affect mosq populations, for ex.

How easy do you think it is to grab this data and join it using lat-long or zipcode to our original dataset?

**Comment by andrew3cooper** [comment]
Looks feasible.  Data run from 2001-18.  I'll need to geocode which could be a hassle with 20k+ rows of data if some don't work on the first geocoding attempt, but I think it'll be do-able.  We might simply count number of vacant properties in the past 1 or 2 years in the block group(s) surrounding each trap or within a specified radius in miles.   We might also need to filter on the type of property violation.  We might end up with multiple versions of this and could use feature selection to choose which ones improve predictions, if any.

**Comment by andrew3cooper** [comment]
I pulled a few examples of google trend data.  Monthly search result data for Chicago on:  dead birds, west nile, west nile symptoms, mosquito bites.  Not sure if it will prove helpful.

**Comment by kapelinskim6** [comment]
Great work Andrew.  I was going to try and use the weather web scraper today if i find some time.  Ill report back if i can produce anything useful.  
Mike
Sent from my T-Mobile 4G LTE Device
-------- Original message --------From: Andrew Cooper <notifications@github.com> Date: 1/28/19  9:48 AM  (GMT-06:00) To: rsangole/capstone_project <capstone_project@noreply.github.com> Cc: kapelinskim6 <kapelinskim@gmail.com>, Mention <mention@noreply.github.com> Subject: Re: [rsangole/capstone_project] Determine new sources of data (#6) 
I pulled a few examples of google trend data.  Monthly search result data for Chicago on:  dead birds, west nile, west nile symptoms, mosquito bites.  Not sure if it will prove helpful.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or mute the thread.
{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/rsangole/capstone_project","title":"rsangole/capstone_project","subtitle":"GitHub repository","main_image_url":"https://github.githubassets.com/images/email/message_cards/header.png","avatar_image_url":"https://github.githubassets.com/images/email/message_cards/avatar.png","action":{"name":"Open in GitHub","url":"https://github.com/rsangole/capstone_project"}},"updates":{"snippets":[{"icon":"PERSON","message":"@andrew3cooper in #6: I pulled a few examples of google trend data.  Monthly search result data for Chicago on:  dead birds, west nile, west nile symptoms, mosquito bites.  Not sure if it will prove helpful.  "}],"action":{"name":"View Issue","url":"https://github.com/rsangole/capstone_project/issues/6#issuecomment-458183943"}}}
[
{
"@context": "http://schema.org",
"@type": "EmailMessage",
"potentialAction": {
"@type": "ViewAction",
"target": "https://github.com/rsangole/capstone_project/issues/6#issuecomment-458183943",
"url": "https://github.com/rsangole/capstone_project/issues/6#issuecomment-458183943",
"name": "View Issue"
},
"description": "View this Issue on GitHub",
"publisher": {
"@type": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]

**Comment by andrew3cooper** [comment]
Re: spraying data, I think this will be out of scope for our project, but we could hypothetically scrape spraying data from Chicago Dept of Public Health news releases.  They identify dates and geographic boundaries, e.g.: 
https://www.chicago.gov/city/en/depts/cdph/provdrs/healthy_communities/news/2018/july/city-to-spray-insecticide-wednesday-to-kill-mosquitoes.html

and 

https://www.chicago.gov/content/dam/city/depts/cdph/food_env/general/West_Nile_Virus/SprayZone_T066_July252018.pdf

**Comment by rsangole** [comment]
> I pulled a few examples of google trend data. Monthly search result data for Chicago on: dead birds, west nile, west nile symptoms, mosquito bites. Not sure if it will prove helpful.

I processed these and are ready in the `processed` folder.

**Comment by rsangole** [comment]
@andrew3cooper I see that you have hospital school and zoning type data in the `raw` now. I'm not familiar with these shape files and how you're processing them to get them ready for consumption. How can I help? Are there scripts I can run? Or some code I need to read? 

Also, I see the NOAA weather data now under `processed`. Do we still need to process the CSV files under `data/raw/chi_weather/Chicago_Only_Weather/` or do we now have enough weather data ?

**Comment by andrew3cooper** [comment]
- Building violations:  I converted the CSV to a shapefile of lat/lng points; I would like to run spatial joins in ArcGIS or QGIS to obtain Census block group, tract, zip code tabulation area and Chicago community area for each building violation.  When done, those fields will be appended to the CSV of building violations, then we can aggregate for a given area (e.g. block group or community area) for a given date range and merge with WNV mosquito trap data for the same area and date range.  There are >1  million building violations, though, and I couldn't get the spatial joins to run successfully yesterday.  Will try again.

- Vacant properties:  I need to geocode these and then do the same kind of spatial joins and aggregation as for building violations.  The idea is to engineer time- and location-specific estimates of unattended properties, which is a proxy for likelihood of having standing water that could breed mosquitoes.

- Zoning: GIS shapefiles that contain polygons with zoning designations; I'll run spatial joins to append zoning designations to WNV mosquito trap data.  This will probably result in a single static zoning variable for each mosquito trap (for modeling purposes) and possibly a set of variables describing each neighborhood (for profiling neighborhoods with increased human risk).  

- Senior centers:  I took the CSV file from Chicago data portal, geocoded it, performed spatial joins, and will reduce it back to a list of senior cities with block group/tract/zcta/community area appended.  we haven't determined exactly how we'll use these data, but probably we'll be aggregating to one or more of the area variables to get a count of senior centers in each area.  Or we might calculate distances from each senior center to each mosquito trap.  I think these data will be used to profile neighborhoods but probably not for modeling.

- Hospitals:  same as senior centers.  I think these data will be used to profile neighborhoods but probably not for modeling.

- Google searches:  I think we'll use these data for modeling.  No prospective model will ever know google trend data until a month later, I think, so we may need to lag data to reflect that.  E.g. if we're predicting July 2018 mosquito data, we might only be able to use google trend data through June 2018.  

- Weather: we'll use these data for modeling.  Need to do feature engineering before merging with mosquito trap data.  Probably we'll take the high/low/average temperature over a few periods of time preceding each mosquito trap date, and the total rainfall and count of wet and dry days or maybe number of consecutive dry days.
